tests/test_http.py

```diff
@@ -47,160 +47,225 @@ def setUp(self):
         # Local test directory
         self.tmpdir = ResourcePath(makeTestTempDir(TESTDIR))
 
-        serverRoot = "www.not-exists.orgx"
         existingFolderName = "existingFolder"
+        notExistingFolderName = "notExistingFolder"
         existingFileName = "existingFile"
         notExistingFileName = "notExistingFile"
 
-        fileForHandleWithRange = "handleWithRange"
-        fileForHandleWithOutRange = "handleWithOutRange"
-
-        self.baseURL = ResourcePath(f"https://{serverRoot}", forceDirectory=True)
-        self.existingFileResourcePath = ResourcePath(
-            f"https://{serverRoot}/{existingFolderName}/{existingFileName}"
+        # DAV endpoint resources
+        self.davEndpoint = "http://dav.not-exists.org"
+        responses.add(
+            responses.OPTIONS,
+            self.davEndpoint,
+            status=200,
+            headers={"DAV": "1,2,3"},
+            auto_calculate_content_length=True,
         )
-        self.notExistingFileResourcePath = ResourcePath(
-            f"https://{serverRoot}/{existingFolderName}/{notExistingFileName}"
+
+        # DAV existing folder and its parent directory
+        self.davExistingFolderResource = ResourcePath(
+            f"{self.davEndpoint}/{existingFolderName}", forceDirectory=True
         )
-        self.existingFolderResourcePath = ResourcePath(
-            f"https://{serverRoot}/{existingFolderName}", forceDirectory=True
+        href = self.davExistingFolderResource.geturl().replace(
```

Comment:
Isn't this code the same as doing `href = self.davExistingFolderResource.relativeToPathRoot` ?

Reply:
Yes, the result is almost the same, although not always. Fixed.